Source/DafnyDriver/CSVTestLogger.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+
+namespace Microsoft.Dafny {
+  public class CSVTestLogger : ITestLoggerWithParameters {
+
+    private readonly List<TestResult> results = new();
+    private TextWriter writer = null;
+
+    public void Initialize(TestLoggerEvents events, string testRunDirectory) {
+    }
+
+    public void Initialize(TestLoggerEvents events, Dictionary<string, string> parameters) {
+      events.TestResult += TestResultHandler;
+      events.TestRunComplete += TestRunCompleteHandler;
+      if (parameters.TryGetValue("LogFileName", out string filename)) {
+        writer = new StreamWriter(filename);
+      }
+    }
+
+    private void TestResultHandler(object sender, TestResultEventArgs e) {
+      results.Add(e.Result);
+    }
+
+    private void TestRunCompleteHandler(object sender, TestRunCompleteEventArgs e) {
+      var realWriter = writer ?? Console.Out;
```

Comment:
I wonder if imitating the TRX logger and defaulting to a generated file name would be better? We could support `csv:LogFileName=-` if we wanted to dump to stdout, and I'm not even sure that's worth it.

Reply:
That does overall seem like the right way to go, though the specific logic the TRX logger uses to choose file names is quite complex. I'll look into approaches that could give us the core benefits of that without hundreds of lines of extra code. :)

Reply:
And now there's some code to auto-generate log file names by default. It uses similar default naming conventions to the TRX logger, but is less configurable.